compiler/rustc_hir/src/hir.rs

```diff
@@ -230,11 +230,28 @@ impl<'hir> PathSegment<'hir> {
 
 #[derive(Clone, Copy, Debug, HashStable_Generic)]
 pub struct ConstArg<'hir> {
-    pub value: &'hir AnonConst,
+    #[stable_hasher(ignore)]
+    pub hir_id: HirId,
```

Comment:
Why is this ignored?

Reply:
Boxy and I were actually talking about this. Some types in HIR ignore their HirId fields and others don’t. There’s no reason this one is ignored other than I was following the pattern of similar types. Do you mind explaining why some are ignored and some aren’t? 😅

Reply:
idk either. I was hoping you did. Will do some archeology

Reply:
FWIW I did find a comment on one of them: https://github.com/rust-lang/rust/blob/565cadb514d35e7b851540edbc172af0f606014f/compiler/rustc_hir/src/hir.rs#L3034-L3036

And this seems to be the commit that first introduced ID ignores for some types; it says it's related to traits, but I'm unclear why `hir::Arm` and `hir::Block`, etc., are relevant to that. e3f9131